src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> linesOfCsvFile = getDataFromCSVFile(fromFileName);
+        int sumOfSupply = Constants.AMOUNT_STARTING_VALUE;
+        int sumOfBuy = Constants.AMOUNT_STARTING_VALUE;
+        if (!linesOfCsvFile.isEmpty()) {
+            String currentType;
+            int currentValue;
+            for (String item : linesOfCsvFile) {
+                String[] itemDataSplit = item.split(Constants.COMA_SEPARATOR);
+                currentType = itemDataSplit[Constants.ITEM_TYPE_INDEX];
+                currentValue = Integer.parseInt(itemDataSplit[Constants.ITEM_VALUE_INDEX]);
```

Parsing the amount with Integer.parseInt(...) may throw NumberFormatException on invalid numeric data. The checklist requires not to ignore exceptions and to provide contextual information. Catch NumberFormatException for the token, and rethrow a RuntimeException that includes the filename and the problematic line so the error is actionable.